config/initializers/active_model_serializers.rb

```diff
@@ -0,0 +1,166 @@
+module AMS
+  module V09
+    class Serializer < ActiveModel::Serializer
+
+       def serializable_hash(adapter_options = nil,
+                             options = {},
+                             adapter_instance = self.class.serialization_adapter_instance)
+         object.nil? ? nil : super
+       end
+    end
+
+    class ArraySerializer < ActiveModel::Serializer::CollectionSerializer
+    end
+
+    class ResourceIdentifier < ActiveModelSerializers::Adapter::JsonApi::ResourceIdentifier
+
+      def initialize(serializer, options)
+        @id   = id_for(serializer)
+        @type = type_for(serializer)
+      end
+
+    private
+
+      def id_for(serializer)
+        serializer.read_attribute_for_serialization(:id)
+      end
+
+    end
+
+    class Base < ActiveModelSerializers::Adapter::Base
+
+      def initialize(serializer, options = {})
+        super
+        @fieldset = options[:fieldset] || ActiveModel::Serializer::Fieldset.new(options.delete(:fields))
+      end
+
+    private
+
+      attr_reader :fieldset
+
+      def attributes_for(serializer, fields)
+        serializer.attributes(fields).except(:id)
+      end
+
+      def resource_object_for(serializer)
+        resource_object = serializer.fetch(self) do
+          resource_object = ResourceIdentifier.new(serializer, instance_options).as_json
+
+          requested_fields = fieldset && fieldset.fields_for(resource_object[:type])
```

Comment:
Style/SafeNavigation: Use safe navigation (`&.`) instead of checking if an object exists before calling the method.